### PR TITLE
feat(config): config file is jinja2-compatible

### PR DESCRIPTION
**Pull request by rdefosse**

### CODE REVIEW

File: docker/Dockerfile.rhel8 [hunk 1]
```diff
@@ -82,6 +82,8 @@ COPY ./rhsm-ca /etc/rhsm/ca
 RUN rm -f /etc/rhsm-host && \
     microdnf update -y && \
     microdnf install -y \
+      python3 \
+      python3-pip \
```
Comment (arora-sagar) [inline docker/Dockerfile.rhel8:86]: @rdefosse We don't need pip lets remove that package and install directly via  python3 and python3-jinja2. I did the same for RHEL and Ubuntu it works fine.

File: etc/spgw_u.conf [hunk 2]
```diff
@@ -57,83 +57,114 @@ SPGW-U =
...
             #PORT                   = 2152;                                     # Default is 2152
             SCHED_PARAMS :
             {
-                #CPU_ID       = 2;
-                SCHED_POLICY = "SCHED_FIFO"; # Values in { SCHED_OTHER, SCHED_IDLE, SCHED_BATCH, SCHED_FIFO, SCHED_RR }
-                SCHED_PRIORITY = @THREAD_S1U_PRIO@;
-                POOL_SIZE = @S1U_THREADS@; # NUM THREADS
+                #CPU_ID        = 2;
+                SCHED_POLICY   = "SCHED_FIFO"; # Values in { SCHED_OTHER, SCHED_IDLE, SCHED_BATCH, SCHED_FIFO, SCHED_RR }
+                SCHED_PRIORITY = {{ env["THREAD_S1U_PRIO"] if "THREAD_S1U_PRIO" in env.keys() else '80' }};
```
Comment (arora-sagar) [inline etc/spgw_u.conf:67]: @rdefosse If possible let set the default values for threading as below

  threadsS1uUl: "1"
  threadsSgiDl: "1"
  threadsSx: "1"
  threadsSgiPrio: 98
  threadsS1uPrio: 88
  threadsSxPrio: 88

File: etc/spgw_u.conf [hunk 3]
```diff
@@ -57,83 +57,114 @@ SPGW-U =
...
+       {%- if env["ENABLE_5G_FEATURES"] is defined %}
        # STRING, {"yes", "no"},
-       ENABLE_5G_FEATURES = "@ENABLE_5G_FEATURES@" # Set to 'yes' to support 5G Features
-       REGISTER_NRF = "@REGISTER_NRF@";            # Set to 'yes' if UPF resgisters to an NRF
-       USE_FQDN_NRF = "@USE_FQDN_NRF@";            # Set to 'yes' if UPF relies on a DNS/FQDN service to resolve NRF's FQDN
-       UPF_FQDN_5G  = "@UPF_FQDN_5G@";             # Set FQDN of UPF 
+       ENABLE_5G_FEATURES = "{{ env["ENABLE_5G_FEATURES"] }}" # Set to 'yes' to support 5G Features
+       REGISTER_NRF = "{{ env["REGISTER_NRF"] }}";            # Set to 'yes' if UPF registers to an NRF
+       USE_FQDN_NRF = "{{ env["USE_FQDN_NRF"] if "USE_FQDN_NRF" in env.keys() else 'no' }}";            # Set to 'yes' if UPF relies on a DNS/FQDN service to resolve NRF's FQDN
+       UPF_FQDN_5G  = "{{ env["UPF_FQDN_5G"] if "UPF_FQDN_5G" in env.keys() else 'none' }}";             # Set FQDN of UPF
```
Comment (arora-sagar) [inline etc/spgw_u.conf:130]: @rdefosse I actually did localhost everywhere if not given. There is a problem to do localhost?

File: etc/spgw_u.conf [hunk 4]
```diff
@@ -57,83 +57,114 @@ SPGW-U =
...
+          ,{ NSSAI_SST = {{ env["NSSAI_SST_1"] }}; NSSAI_SD = "{{ env["NSSAI_SD_1"] }}";  DNN_LIST = ({DNN = "{{ env["DNN_1"] }}";}); }
+          {%- endif %}
+          {%- if env["NSSAI_SST_2"] is defined %}
+          ,{ NSSAI_SST = {{ env["NSSAI_SST_2"] }}; NSSAI_SD = "{{ env["NSSAI_SD_2"] }}";  DNN_LIST = ({DNN = "{{ env["DNN_2"] }}";}); }
+          {%- endif %}
+          {%- if env["NSSAI_SST_3"] is defined %}
+          ,{ NSSAI_SST = {{ env["NSSAI_SST_3"] }}; NSSAI_SD = "{{ env["NSSAI_SD_3"] }}";  DNN_LIST = ({DNN = "{{ env["DNN_3"] }}";}); }
+          {%- endif %}
+          {%- else %}
+          { NSSAI_SST = 1; NSSAI_SD = "0xFFFFFF";  DNN_LIST = ({DNN = "default";}); }
```
Comment (arora-sagar) [inline etc/spgw_u.conf:166]: @rdefosse if you want to mention a default slice, then mention DNN `oai` please

File: docker/Dockerfile.rhel8 (hunk 1 above)
Comment (rdefosse) [reply docker/Dockerfile.rhel8:86]: My method does not require any subscription. And `python3-pip` is included when installing `python3`. Just making sure.

**Comment by arora-sagar** [comment]
Rest all good thanks

File: etc/spgw_u.conf (hunk 3 above)
Comment (rdefosse) [reply etc/spgw_u.conf:130]: @arora-sagar 

That's for me the bug in `NRF` --> if `USE_FQDN-NRF` is set to `no` (default), the `UPF_FQDN_5G` should be `garbage`.

But OK. I will put `localhost`

File: docker/Dockerfile.rhel8 (hunk 1 above)
Comment (arora-sagar) [reply docker/Dockerfile.rhel8:86]: Okay

**Review by arora-sagar — APPROVED** [review]

**Pull request merged**